aten/src/ATen/benchmarks/tensor_add.cpp

```diff
@@ -0,0 +1,30 @@
+#include <ATen/ATen.h>
+
+#include <benchmark/benchmark.h>
+
+static void tensor_add(benchmark::State& state) {
+  const size_t batchSize = static_cast<size_t>(state.range(0));
+  const size_t channels = static_cast<size_t>(state.range(1));
+
+  at::Tensor a = at::rand({batchSize, channels});
+  at::Tensor b = at::rand({batchSize, channels});
+  at::Tensor c;
+  for (auto _ : state) {
```

Comment:
I'm not sure whether this runs, for a benchmark, the risk of getting optimized out considering the contents of this loop are invariant.

Reply:
You mean the entire loop is optimized away?
This is how you are supposed to use google benchmark it seems: https://github.com/google/benchmark. Also qnnpack benchmarks are similarly written.
Also the next PR in the stack has the result of running this benchmark. From that it does not seem to be case.